packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { AnyFunctionWithRecord, IdempotencyRecordStatus } from './types';
+import { IdempotencyOptions } from './IdempotencyOptions';
+import { IdempotencyRecord } from 'persistence';
+import { IdempotencyInconsistentStateError, IdempotencyItemAlreadyExistsError, IdempotencyAlreadyInProgressError, IdempotencyPersistenceLayerError } from './Exceptions';
+
+export class IdempotencyHandler<U> {
+
+  public constructor(private functiontoMakeIdempotent: AnyFunctionWithRecord<U>, private functionPayloadToBeHashed: unknown, 
+    private idempotencyOptions: IdempotencyOptions, private fullFunctionPayload: Record<string, any>) {}
+
+  public determineResultFromIdempotencyRecord(idempotencyRecord: IdempotencyRecord): Promise<U> | U{ 
+    if (idempotencyRecord.getStatus() === IdempotencyRecordStatus.EXPIRED) {
+      throw new IdempotencyInconsistentStateError();
+    } else if (idempotencyRecord.getStatus() === IdempotencyRecordStatus.INPROGRESS){
+      throw new IdempotencyAlreadyInProgressError(`There is already an execution in progress with idempotency key: ${idempotencyRecord.idempotencyKey}`);
+    } else {
+      // Currently recalling the method as this fulfills FR1. FR3 will address using the previously stored value https://github.com/awslabs/aws-lambda-powertools-typescript/issues/447
+      return this.functiontoMakeIdempotent(this.fullFunctionPayload); 
```

Comment:
Does this achieve idempotency (FR1) or does it just prevent concurrent calls from executing the same code?

Reply:
The main question I have is: should we recall the method or should we just not return a response. I don't recall whether we discussed this earlier.